### PR TITLE
feat: build date-time, bumping after build; edit error logging

### PR DESCRIPTION
**Pull request by EttyKitty**
### Build date-time:
- A separate json file that tracks version number and build date-time.
- Build date-time and version as globals.
- Build date-time bump after building with github action.
- New `json_to_gamemaker()` function, that converts json files to dslists or structs.

### Logging:
- `scr_log.gml` refactors and edits.
- New `handle_exception()` function, that accepts an exception struct from a try-catch loop and handles it as `try_and_report_loop()` would. Just a light function to put in all catch blocks to display player error message and log it into file.
- A couple of additional macros to reduce typo chance in strings.
- JSDoc for some functions.
- A TODO comment about a possible implementation of session logs to replace bloated `debug_log.ini`.

### Misc:
- `format_stacktrace()` renamed into `array_to_string_list()`, because that's what it does.
- `markdown_codeblock()` to wrap a string in markdown codeblock symbols.

### Notes:
- ~Build date-time is hidden from game menu for now, to avoid confusion while it's unused.~

### CODE REVIEW

File: scripts/scr_json_functions/scr_json_functions.gml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+/// @param {string} _json_path full path to the file.
+/// @param {function} _func json_parse or json_decode, to get a struct or a dslist.
+function json_to_gamemaker(_json_path, _func) {
+    try {
+        if (file_exists(_json_path)){
+            var _file = file_text_open_read(_json_path);
+            var _json_string = "";
+            var _parsed_json = {};
+    
+            while (!file_text_eof(_file)) {
```
Comment (OH296) [inline scripts/scr_json_functions/scr_json_functions.gml:12]: i presume there is no way a text file could end up having lines being added making this an infinite loop? (always worth asking when it comes to while loops lol)

File: scripts/scr_log/scr_log.gml [hunk 2]
```diff
@@ -52,45 +59,65 @@ function try_and_report_loop(dev_marker="generic crash",func, turn_end=true, arg
...
-    var _formatted_stacktrace = "";
-
-    if (!is_array(stacktrace)) {
-        return;
+/// @function markdown_codeblock
+/// @description Puts codeblock symbols around a string.
+/// @param {string} _message
+/// @return {string}
+function markdown_codeblock(_message) {
+    if (string_length(_message > 0)) {
```
Comment (OH296) [inline scripts/scr_log/scr_log.gml]:
```suggestion
    if (string_length(_message) > 0) {
```

**Review by OH296 — CHANGES_REQUESTED** [review]
one minor error, otherwise all seems good

File: scripts/scr_json_functions/scr_json_functions.gml (hunk 1 above)
Comment (EttyKitty) [reply scripts/scr_json_functions/scr_json_functions.gml:12]: @OH296,
I don't think so?
And I'm not sure if it's possible to failsafe this without an overhead. Especially if the input file will be big.

File: scripts/scr_log/scr_log.gml (hunk 2 above)
Comment (EttyKitty) [reply scripts/scr_log/scr_log.gml]: @OH296,
is this an error?

File: scripts/scr_json_functions/scr_json_functions.gml (hunk 1 above)
Comment (OH296) [reply scripts/scr_json_functions/scr_json_functions.gml:12]: not a big issue probably fine

**Pull request merged**